File: joao pedro/Problema0380/Problema0380.java
import java.util.Scanner;

public class Problema0380 {
    public static void main(String[] args) {
        
        Scanner input=new Scanner(System.in);
        int [][]matriz=new int[4][5];
        double [] vetor=new double[4];
        int []quantidadeArmazem=new int[5];
        int []quantidadeProd=new int[5];
       int maiorEstoque=0;
       int posicaoPreco=0;
       int []valorCada=new int[5];
        int menorEstoque=Integer.MAX_VALUE;
            for(int j=0;j<4;j++){
                System.out.println("Digite o preco do produto do produto"+j+": ");
                vetor[j]=Double.valueOf(input.nextLine());

            }
            for(int i=0;i<5;i++){
                for(int j=0;j<4;j++){
                    System.out.println("Digite a quantidade do produto "+i+" no armazem "+j+": ");
                    matriz[j][i]=Integer.valueOf(input.nextLine());
                    quantidadeArmazem[i]=matriz[j][i]+quantidadeArmazem[i];
                    quantidadeProd[j]=matriz[j][i]+quantidadeProd[j];
                    if(matriz[j][i]>maiorEstoque){
                        maiorEstoque=matriz[j][i];
                        posicaoPreco=j;
                    }
                    if(matriz[j][i]<menorEstoque){
                        menorEstoque=matriz[j][i];
                    }
                }
            }
        for(int i=0;i<5;i++){
            System.out.println("Quantidade no armazem "+i+": "+quantidadeArmazem[i]);

        }
        for(int i=0;i<4;i++){
            System.out.println("Quantidade do produto "+i+": " +quantidadeProd[i]);
        }
        System.out.println("Preco do produto de maior estoque: "+vetor[posicaoPreco]);
        System.out.println("Menor estoque armazenado: "+menorEstoque);
       
    }
}

File: joao pedro/Problema0380/Problema0380.py
matriz = [[0] * (5) for _ in range(4)]
matrizx=0
vetor = [0.0] * (4)
vetorx=0
quantidadeArmazem = [0] * (5)
quantidadeProd = [0] * (5)
maiorEstoque = 0
posicaoPreco = 0
valorCada = [0] * (5)
menorEstoque = 9999999
j = 0
while (j < 4) :
    print("Digite o preco do produto do produto" ,j, ": ")
    vetorx=float(input(""))
    vetor.insert(j,vetorx)
    j += 1
i = 0
while (i < 5) :
    
    while (j < 4) :
        print("Digite a quantidade do produto " + str(i) + " no armazem " + str(j) + ": ")
        matrizx=int(input(""))
        matriz[i].insert(j,matrizx)
        quantidadeArmazem[i] = matriz[j][i] + quantidadeArmazem[i]
        quantidadeProd[j] = matriz[j][i] + quantidadeProd[j]
        if matrizx > maiorEstoque :
            maiorEstoque = matriz[j][i]
            posicaoPreco = j
        if matrizx < menorEstoque :
                menorEstoque = matriz[j][i]
        j += 1
    i += 1
    j=0
i = 0
while (i < 5) :
    print("Quantidade no armazem " + str(i) + ": " + str(quantidadeArmazem[i]))
    i += 1
i = 0
while (i < 4) :
    print("Quantidade do produto " + str(i) + ": " + str(quantidadeProd[i]))
    i += 1
print("Preco do produto de maior estoque: " + str(vetor[posicaoPreco]))
print("Menor estoque armazenado: " + str(menorEstoque))